app/forms/candidate_interface/english_gcse_grade_form.rb

```diff
@@ -0,0 +1,278 @@
+            params[:grade_english_studies_single] = grades['english_studies_single_award']
+            params[:english_studies_single_award] = true
+          end
+
+          if grades.keys.include? 'english_studies_double_award'
+            english_gcses << 'english_studies_double_award'
+            params[:grade_english_studies_double] = grades['english_studies_double_award']
+            params[:english_studies_double_award] = true
+          end
+
+          other_english_gcse_name = grades.keys.select { |k|
+            %w[
+              english_single_award english_double_award
+              english_language
+              english_literature
+              english_studies_single_award english_studies_double_award
+            ].exclude? k
+          } .first
+
+          english_gcses << 'other_english_gcse' if other_english_gcse_name
+          params[:english_gcses] = english_gcses
+
+          params[:other_english_gcse] = other_english_gcse_name
+          params[:other_english_gcse_name] = other_english_gcse_name if other_english_gcse_name
+          params[:grade_other_english_gcse] = grades[other_english_gcse_name] if other_english_gcse_name
+        end
+
+        params
+      end
+    end
+
+    def assign_values(params)
+      english_gcses = params[:english_gcses]
+      self.english_gcses = english_gcses
+
+      self.english_single_award = english_gcses.include? 'english_single_award'
+      self.grade_english_single = params[:grade_english_single]
+
+      self.english_double_award = english_gcses.include? 'english_double_award'
+      self.grade_english_double = params[:grade_english_double]
+
+      self.english_language = english_gcses.include? 'english_language'
+      self.grade_english_language = params[:grade_english_language]
+
+      self.english_literature = english_gcses.include? 'english_literature'
+      self.grade_english_literature = params[:grade_english_literature]
+
+      self.english_studies_single_award = english_gcses.include? 'english_studies_single_award'
+      self.grade_english_studies_single = params[:grade_english_studies_single]
+
+      self.english_studies_double_award = english_gcses.include? 'english_studies_double_award'
+      self.grade_english_studies_double = params[:grade_english_studies_double]
+
+      self.other_english_gcse = english_gcses.include? 'other_english_gcse'
+      self.other_english_gcse_name = params[:other_english_gcse_name]
+      self.grade_other_english_gcse = params[:grade_other_english_gcse]
+      self
+    end
+
+    def save_grades
+      grades = build_grades_json
+      if !valid?(:grades)
+        log_validation_errors(:grades)
+        return false
+      end
+      qualification.update(grades: grades, grade: nil)
+    end
+
+    def save_grade
+      if !valid?(:grade)
+        log_validation_errors(:grade)
+        return false
+      end
+      qualification.update(grade: set_grade, award_year: award_year)
+    end
```

Comment:
Again, this will be refactored as part of the next ticket so we an have just the one `save` method instead. Similar to the final state of Toby's `ScienceGcseGradeForm`